includes/reader-activation/class-reader-activation.php

```diff
@@ -2359,5 +2354,31 @@ private static function get_logged_in_reader_email_address() {
 
 		return $email_address;
 	}
+
+
+	/**
+	 * Log in a reader.
+	 *
+	 * @param WP_User $user WP_User object.
+	 */
+	private static function log_in_reader( $user ) {
+		\wp_clear_auth_cookie();
+		\wp_set_current_user( $user->ID );
+		\wp_set_auth_cookie( $user->ID, true );
+		\do_action( 'wp_login', $user->user_login, $user );
+		Logger::log( 'Logged in user ' . $user->ID );
+	}
```

Comment:
> Why not use set_current_reader()?

I wanted to avoid the redundant validation and potential returning of WP_Error that comes with `set_current_reader` since in this case we are always going to be logging in the user regardless of whether they are a `reader` or not at successful password reset.

That said, I'm fine going with `set_current_reader` instead if you prefer! Updated in https://github.com/Automattic/newspack-plugin/pull/3510/commits/bfe042f16fe1bd115c788f86799bfbf58da37d14

Reply:
> in this case we are always going to be logging in the user regardless of whether they are a reader or not at successful password reset.

To be safe, it's better to have this only for readers, which I believe is how it's now after bfe042f16fe1bd115c788f86799bfbf58da37d14. Publishers may have additional layers of security added to the login form aimed at users with higher capabilities that this could bypass (IP/proxy, brute force protection, 2fa, etc.). A bad actor could find a way to reset an account's password, but if there's additional protection, that doesn't mean they're allowed in instantly.

That's also why I hesitate to have a method that doesn't check if it's a reader before logging in. It could get accidentally misused.